node/impl/full/eth.go

```diff
@@ -1455,77 +1455,23 @@ func lookupEthAddress(ctx context.Context, addr address.Address, sa StateAPI) (e
 }
 
 func newEthTxFromFilecoinMessage(ctx context.Context, smsg *types.SignedMessage, sa StateAPI) (ethtypes.EthTx, error) {
-	fromEthAddr, err := lookupEthAddress(ctx, smsg.Message.From, sa)
-	if err != nil {
-		return ethtypes.EthTx{}, err
-	}
-
-	toEthAddr, err := lookupEthAddress(ctx, smsg.Message.To, sa)
+	tx, err := ethtypes.EthTxFromSignedMessage(smsg)
 	if err != nil {
-		return ethtypes.EthTx{}, err
+		return ethtypes.EthTx{}, xerrors.Errorf("failed to convert from signed message: %w", err)
 	}
 
-	toAddr := &toEthAddr
-	input := smsg.Message.Params
-	// Check to see if we need to decode as contract deployment.
-	// We don't need to resolve the to address, because there's only one form (an ID).
-	if smsg.Message.To == builtintypes.EthereumAddressManagerActorAddr {
```

Comment:
So, the downside of unifying this is that we'll no longer be able to interpret "native" messages as ethereum transactions. On the other hand... that's probably not desirable anyways.